docs/src/architecture/accounts.md

```diff
@@ -8,44 +8,53 @@ An account is an entity which holds assets and defines rules of how these assets
 In the above:
 
 * **Account ID** is a unique identifier (8 bytes) of an account which does not change throughout its lifetime. It also specifies the type of the underlying account. 
-* **Storage** is user-defined data which can be stored in an account. The data is stored as key-value map which maps 256-bit keys to 256-bit values.
-* **Nonce** is the account's nonce increments every time the account changes its state. 
+* **Storage** is user-defined data which can be stored in an account. 256 values of four field elements each can be stored in a key-value map mapping four field element keys to four field element values.
+* **Nonce** is a counter that increments every time the account changes its state. The counter is monotonically increasing. 
```

Comment:
I'd change this to: "... is a counter which must be incremented whenever account state changes. Nonce values must be monotonically increasing, but can be incremented by any value smaller than $2^{32}$ for every account update."

Reply:
I shortened it in the bullet list and gave an in-depth description using your sentence in a short separate section. 

To me the addition of "Nonce values must be monotonically increasing, but can be incremented by any value smaller than 2^32 for every account update." doesn't add value. This seems to be the definition of a counter which is always monotonically increasing (actually to be 100% correct it must be strictly monotonically increasing).